classes/meeting.php

```diff
@@ -51,6 +51,60 @@ public function __construct(instance $instance) {
         $this->instance = $instance;
     }
 
+    /**
+     * Helper to create and join a meeting
+     *
+     * @param instance $instance
+     * @param int $origin
+     * @return object|void
+     * @throws \coding_exception
+     * @throws \core\invalid_persistent_exception
+     * @throws bigbluebutton_exception
+     */
+    public static function create_and_join_meeting(instance $instance, $origin = logger::ORIGIN_INDEX): array {
```

Comment:
Wouldn't this make more sense to be in the instance and to return a url only?

If it's a failure, have it throw an exception rather than a weird array situation.

Reply:
What i'm saying here is that the function should have signature:

```
public static function join_Meeting(instance $instance, int $origin = logger::ORIGIN_INDEX): string;
```

* There is no need for the distinction between `join` and `create and join` here. That is an implementation detail
* The URL is never used if there is any error or warning code so don't return it - throw an exception.

i.e.

```
$meeting = new meeting($instance);

if (!$meeting->is_running()) {
    if ($instance->user_must_wait_to_join()) {
        throw new \mod_bigbluebuttonbn\cannot_join_exception('waitformoderator');
    }

    // Create the meeting.
    $response = $meeting->create_meeting();

    // TODO Maybe move the is_recorded() stuff to the create_meeting() function??
    logger::log_meeting_created_event($instance);
}

if ($instance->has_user_limit_been_reached($meeting->get_participant_count()) {
    throw new \mod_bigbluebuttonbn\cannot_join_exception('userlimitreached');
}

return $meeting->get_join_url($origin);
```


Other things to consisder:
* move the recording creation to the create_meeting function
*